profilehooks.py

```diff
@@ -750,10 +771,28 @@ def __call__(self, *args, **kw):
                 funcname = fn.__name__
                 filename = fn.__code__.co_filename
                 lineno = fn.__code__.co_firstlineno
-                sys.stderr.write("\n  %s (%s:%s):\n    %.3f seconds\n\n" % (
-                    funcname, filename, lineno, duration
-                ))
-                sys.stderr.flush()
+                args_str = ', '.join(str(item) for item in args)
+                kw_str = ', '.join(['%s=%s' % (key, value) for key, value in kw.items()])
+                arguments_str = args_str
+                if kw_str:
+                    if args_str:
+                        arguments_str += ', '
+                    arguments_str += kw_str
+                message = "%s:line %s:%s(%s):%.3f seconds" % (
+                    filename,
+                    lineno,
+                    funcname,
+                    arguments_str,
+                    duration
+                )
+                if self.logger:
+                    self.logger.log(self.log_level, message)
+                else:
+                    message = "%s (%s:%s):\n    %.3f seconds\n\n" % (
+                        funcname, filename, lineno, duration
```

Comment:
Maybe it's an opportunity to make this more readable by using str.format() instead of str.\__mod__?

```python
    sys.stderr.write("\n  {funcname} ({filename}:{lineno}\n    {duration:.3f} secons\n\n".format(
        funcname=funcname, filename=filename, lineno=lineno, duration=duration,
    ))
    sys.stderr.flush()
```